server/Arcadia.Assistant.CSP/Vacations/ArcadiaVacationRegistry.cs

```diff
@@ -66,12 +81,107 @@ protected override void OnReceive(object message)
 
                     break;
 
+                case CalendarEventCreated msg when msg.Event.Type == CalendarEventTypes.Vacation:
+                    this.UpsertVacation(msg.Event, null, new string[0])
+                        .PipeTo(
+                            this.Self,
+                            success: () => VacationPersistSuccess.Instance,
+                            failure: e => new VacationPersistFailed(e.Message));
+                    break;
+
+                case CalendarEventChanged msg when msg.NewEvent.Type == CalendarEventTypes.Vacation:
+                    this.UpsertVacation(msg.NewEvent, msg.OldEvent, new string[0])
+                        .PipeTo(
+                            this.Self,
+                            success: () => VacationPersistSuccess.Instance,
+                            failure: e => new VacationPersistFailed(e.Message));
+                    break;
+
+                case CalendarEventApprovalsChanged msg when msg.Event.Type == CalendarEventTypes.Vacation:
```

Comment:
Put all this code in a separate child Actor? Why add a hundred LOC in existing one where it doesn't use its features?

Reply:
Because you said to add them into existing actor, didn't you? ;)

Reply:
Done.